bleep-core/src/scala/bleep/commands/Test.scala

```diff
@@ -4,10 +4,23 @@ package commands
 import bleep.bsp.BspCommandFailed
 import bleep.internal.{DoSourceGen, TransitiveProjects}
 import ch.epfl.scala.bsp4j
+import scala.jdk.CollectionConverters.*
 
 import bloop.rifle.BuildServer
+import cats.data.NonEmptyList
 
-case class Test(watch: Boolean, projects: Array[model.CrossProjectName]) extends BleepCommandRemote(watch) with BleepCommandRemote.OnlyChanged {
+case class Test(watch: Boolean, projects: Array[model.CrossProjectName], testSuites: Option[NonEmptyList[String]])
+    extends BleepCommandRemote(watch)
+    with BleepCommandRemote.OnlyChanged {
+
+  private def intersectTestSuites(fromBuild: List[String], fromCli: List[String]): List[String] =
+    fromBuild.filter { fullyQualifiedCls =>
+      fromCli.exists { fromCliCls =>
+        val fromBuildFqn = fullyQualifiedCls.split('.')
+        val fromCliFqn = fromCliCls.split('.')
+        fromBuildFqn.sameElements(fromCliFqn) || fromBuildFqn.last == fromCliFqn.last
```

Comment:
I don't really like this predicate here. 

to start: can we make both lists, `fromBuild` and `fromCli` lowercase before we start filtering?
then let's have one rule instead of two. one of the following maybe:

- check if `fromCliFqn` is a substring of  `fromBuildFqn` 
- let `fromCliFqn` be a regex and see if it matches
- let `fromCliFqn` be a glob and see if it matches
or a rule like that.

then document it, at least with a description in the CLI interface

Reply:
I initially went with the substring approach, but after testing it some I really did not like it as I felt it became too easy for tests to clash, e.g RepositoryTest is a substring of IntegrationRepositoryTest. So my idea here was to make it as predictable as possible. It either runs whatever suites has that name e.g AppTest will run foo.AppTest and bar.AppTest and if you want to pick one you specify with fully qualified name. 

I will add documentation to the CLI and bleep.build page once the predicate is decided :)